src/Processor/Braintree/Mapper/Subscription/ModifierMapper.php

```diff
@@ -54,10 +54,13 @@ private function withNewDefaultModifiers(array &$request, array $newModifiers):
                 $default = Arr::replaceKeys($newDefault->toArray(), ['id' => 'inheritedFromId']);
 
                 if (is_array($modifiers)) {
-                    // if new modifier is in default, merge modifiers
-                    $userSuppliedModifiers = Arr::filter($modifiers, fn ($modifier) => $modifier['inheritedFromId'] !== $default['inheritedFromId']);
+                    $otherUserSuppliedModifiers = Arr::filter($modifiers, fn ($modifier) => $modifier['inheritedFromId'] !== $default['inheritedFromId']);
+                    $defaultModifierWithProvidedUpdates = [array_merge(
+                        $default,
+                        ...Arr::filter($modifiers, fn ($modifier) => $modifier['inheritedFromId'] === $default['inheritedFromId']) ?? []
+                    )];
 
-                    return [...$userSuppliedModifiers, $default];
+                    return array_merge($otherUserSuppliedModifiers, $defaultModifierWithProvidedUpdates);
```

Comment:
Same question as above. Does the spread operator work here?

Reply:
Sorta a reply to both...

We're using them differently because the spread doesn't handle string keys... so we need to use array merge for those. If we think the interchanging use of them might be confusing, best would be to use array_merge for all the cases, in my opinion.

We can use `...Arr::filter(...)` because the return of that is 
```php
Array(
[0] => [
    'inheritedFromId' => 'foo',
    'quantity' => 8
])
```

So the spread is essentially just giving us the first result. Perhaps that might even be better as `$array[0]` instead of the spread?

What say ye?

Reply:
sounds good to me man

Reply:
💻